chunk/schema_util.go

```diff
@@ -0,0 +1,116 @@
+package chunk
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/prometheus/common/model"
+)
+
+func buildRangeKey(ss ...[]byte) []byte {
+	length := 0
+	for _, s := range ss {
+		length += len(s) + 1
+	}
+	output, i := make([]byte, length, length), 0
+	for _, s := range ss {
+		copy(output[i:i+len(s)], s)
+		i += len(s) + 1
+	}
+	return output
+}
+
+func deconstructRangeKey(value []byte) [][]byte {
```

Comment:
👍 but can we either call `buildRangeKey` `constructRangeKey` or change this to `unbuildRangeKey`?  Or perhaps call `buildRangeKey` `mantleRangeKey` and make this one `dismantleRangeKey`?